### PR TITLE
[Tests][Unit] Complete Bounties unit test for frontend/app/src/store/main.ts

### LINKED ISSUE #1230: [Tests][Unit] Complete Bounties unit test for frontend/app/src/store/main.ts
### Context

main.ts houses all the API call functions. Add tests to complete all the bounty API calls,

File to add test

`src/store/__test__/main.spec.ts`

Make Mock APIs to get:
 
- To delete a bounty,  check ```deleteBounty``` in  frontend/app/src/store/main.ts, and mock the request API request in the function.
- To remove a bounty assignee, check ```deleteBountyAssignee``` in  frontend/app/src/store/main.ts, and mock the request API request in the function.
- To update a bounty payment status, check ```updateBountyPaymentStatus``` in  frontend/app/src/store/main.ts, and mock the request API request in the function.
- To get a bounty by its id, check ```getBountyById``` in  frontend/app/src/store/main.ts, and mock the request API request in the function.
- To get a bounty Index in the database by its id, check ```getBountyIndexById``` in  frontend/app/src/store/main.ts, and mock the request API request in the function.

Check ```src/store/__test__/main.spec.ts`` for example

We want to test that Bounty APIs return the right data

### Acceptance Criteria

- [ ] Test that the tests pass with the right data
- [ ] Also test for failures.

### PR DESCRIPTION
**Pull request by wright-eric**
## Describe your changes
Added unit test for bounties function in store.
used sinon to mock api calls

## Issue ticket number and link
closes #1230 

## Type of change
unit tests

Please delete options that are not relevant.

- [x] added tests for the function ``deleteBounty``
- [x] added tests for the function ``deleteBountyAssignee``
- [x] added tests for the function ``updateBountyPaymentStatus``
- [x] added tests for the function ``getBountyById``
- [x] added tests for the function ``getBountyIndexById``

## Checklist before requesting a review
- [x] I have performed a self-review of my code

### CODE REVIEW

**Comment by wright-eric** [comment]
cc @ecurrencyhodler  @elraphty

**Comment by wright-eric** [comment]
@elraphty  @kevkevinpal  i have also added the unit test for the functioned mentioned in the issue #955  for signed out perspective, let me know if i need to add other test.

**Comment by ecurrencyhodler** [comment]
@wright-eric we generally want one pr per issue. But I'll let kev or raph decide whether that's okay for this one because they're unit tests.

**Comment by elraphty** [comment]
@wright-eric please create a separate PR for the signed-out unit tests.

File: frontend/app/src/store/__test__/main.spec.ts
Comment (wright-eric) [reply frontend/app/src/store/__test__/main.spec.ts:110]: I am just trying to mock rhe scenario where even though we send correct data the server responds with exception maybe the server stopped or there was some internal error in the server.

Comment (kevkevinpal) [reply frontend/app/src/store/__test__/main.spec.ts:110]: I think this test makes sense, its testing if the API throws an error what happens

**Review by kevkevinpal — APPROVED** [review]
These changes look good to me! nice work

**Comment by wright-eric** [comment]
> @wright-eric please create a separate PR for the signed-out unit tests.

i have removed the test for the signed out perspective. will create a separate PR for that after this is merged because the changes are in same file and would cause conflicts.

**Comment by ecurrencyhodler** [comment]
paid!

**Pull request merged**
